packages/next/client/page-loader.js

```diff
@@ -112,30 +133,33 @@ export default class PageLoader {
     })
   }
 
-  async loadScript (route) {
+  async loadRoute (route) {
     await this.promisedBuildId
 
     route = this.normalizeRoute(route)
     let scriptRoute = route === '/' ? '/index.js' : `${route}.js`
 
-    const script = document.createElement('script')
+    const url = `${this.assetPrefix}/_next/static/${encodeURIComponent(
+      this.buildId
+    )}/pages${scriptRoute}`
+    this.loadScript(url, route, true)
+  }
 
+  loadScript (url, route, isPage) {
+    const script = document.createElement('script')
     if (process.env.__NEXT_MODERN_BUILD && 'noModule' in script) {
       script.type = 'module'
-      scriptRoute = scriptRoute.replace(/\.js$/, '.module.js')
+      // Only page bundle scripts need to have .module added to url,
+      // dependencies already have it added during build manifest creation
+      if (isPage) url = url.replace(/\.js$/, '.module.js')
```

Comment:
I switched to this because I wanted the code to be a little more explicit about why sometimes we need to add the .module.js and sometimes we don't. Just applying the logic based on whether or not it's already there seems like it could obfuscate the source of bugs in the future